ee/modules/502-delivery/templates/argocd/repo-server/argocd-repo-server-deployment.yaml

```diff
@@ -0,0 +1,335 @@
+              valueFrom:
+                configMapKeyRef:
+                  key: reposerver.default.cache.expiration
+                  name: argocd-cmd-params-cm
+                  optional: true
+            - name: ARGOCD_REPO_SERVER_OTLP_ADDRESS
+              valueFrom:
+                configMapKeyRef:
+                  key: otlp.address
+                  name: argocd-cmd-params-cm
+                  optional: true
+            - name: ARGOCD_REPO_SERVER_MAX_COMBINED_DIRECTORY_MANIFESTS_SIZE
+              valueFrom:
+                configMapKeyRef:
+                  key: reposerver.max.combined.directory.manifests.size
+                  name: argocd-cmd-params-cm
+                  optional: true
+            - name: ARGOCD_REPO_SERVER_PLUGIN_TAR_EXCLUSIONS
+              valueFrom:
+                configMapKeyRef:
+                  key: reposerver.plugin.tar.exclusions
+                  name: argocd-cmd-params-cm
+                  optional: true
+            - name: ARGOCD_REPO_SERVER_ALLOW_OUT_OF_BOUNDS_SYMLINKS
+              valueFrom:
+                configMapKeyRef:
+                  key: reposerver.allow.oob.symlinks
+                  name: argocd-cmd-params-cm
+                  optional: true
+            - name: ARGOCD_REPO_SERVER_STREAMED_MANIFEST_MAX_TAR_SIZE
+              valueFrom:
+                configMapKeyRef:
+                  key: reposerver.streamed.manifest.max.tar.size
+                  name: argocd-cmd-params-cm
+                  optional: true
+            - name: ARGOCD_REPO_SERVER_STREAMED_MANIFEST_MAX_EXTRACTED_SIZE
+              valueFrom:
+                configMapKeyRef:
+                  key: reposerver.streamed.manifest.max.extracted.size
+                  name: argocd-cmd-params-cm
+                  optional: true
+            - name: HELM_CACHE_HOME
+              value: /helm-working-dir
+            - name: HELM_CONFIG_HOME
+              value: /helm-working-dir
+            - name: HELM_DATA_HOME
+              value: /helm-working-dir
+          image: {{ include "helm_lib_module_image" (list . "argocd") }}
+          imagePullPolicy: Always
+          livenessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: /healthz?full=true
+              port: 8084
+            initialDelaySeconds: 30
+            periodSeconds: 5
+          name: argocd-repo-server
+          ports:
+            - containerPort: 8081
+            - containerPort: 8084
+          resources:
+            requests:
+            {{- include "helm_lib_module_ephemeral_storage_logs_with_extra" 1024 | nindent 14 }}
+{{- if not ( .Values.global.enabledModules | has "vertical-pod-autoscaler-crd") }}
+            {{- include "resources_argocd-repo-server" . | nindent 14 }}
+{{- end }}
+          readinessProbe:
+            httpGet:
+              path: /healthz
+              port: 8084
+            initialDelaySeconds: 5
+            periodSeconds: 10
+          securityContext:
```

Comment:
There is a helper for security context.

Reply:
Argo CD specified defaults for containers explicitly

```yaml
securityContext:
  allowPrivilegeEscalation: false   # convered by helm_lib_module_container_security_context_read_only_root_filesystem_capabilities_drop_all
  capabilities:                     # convered by helm_lib_module_container_security_context_read_only_root_filesystem_capabilities_drop_all
    drop:                           # convered by helm_lib_module_container_security_context_read_only_root_filesystem_capabilities_drop_all
      - ALL                         # convered by helm_lib_module_container_security_context_read_only_root_filesystem_capabilities_drop_all
  readOnlyRootFilesystem: true      # convered by helm_lib_module_container_security_context_read_only_root_filesystem_capabilities_drop_all
  runAsNonRoot: true                # what about it?
  seccompProfile:                   # what about it?
    type: RuntimeDefault            # what about it?
```

What should I do?

1. Create another helper with these defaults
2. Specify these defaults in `helm_lib_module_container_security_context_read_only_root_filesystem_capabilities_drop_all`
3. Ignore these fields as they seem to be default and they don't overwrite what is in the Pod security context

Reply:
3